lib/bunch/kv_list.ex

```diff
@@ -36,4 +36,97 @@ defmodule Bunch.KVList do
   def map_values(list, f) do
     list |> Enum.map(fn {key, value} -> {key, f.(value)} end)
   end
+
+  @doc """
+  Filters elements of `list` by keys using function `f`.
+
+  ## Example
+
+      iex> #{inspect(__MODULE__)}.filter_by_keys([a: 1, b: 2, a: 3], & &1 == :a)
+      [a: 1, a: 3]
+
+  """
+  @spec filter_by_keys(t(k, v), (k -> as_boolean(term))) :: t(k, v) when k: any, v: any
+  def filter_by_keys(list, f) do
+    list |> Enum.filter(&apply_to_key(&1, f))
+  end
+
+  @doc """
+  Filters elements of `list` by values using function `f`.
+
+  ## Example
+
+      iex> #{inspect(__MODULE__)}.filter_by_values([a: 1, b: 2, a: 3], & &1 |> rem(2) == 0)
+      [b: 2]
+
+  """
+  @spec filter_by_values(t(k, v), (v -> as_boolean(term))) :: t(k, v) when k: any, v: any
+  def filter_by_values(list, f) do
+    list |> Enum.filter(&apply_to_value(&1, f))
+  end
+
+  @doc """
+  Executes `f` with each key from list.
```

Comment:
Rather executes `f` for each key in list, see: https://hexdocs.pm/elixir/Enum.html#each/2 for reference